src/axom/inlet/Field.cpp

```diff
@@ -58,12 +60,168 @@ bool Field::required()
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    if (m_docEnabled) {
+      m_sidreGroup->createViewScalar("defaultValue", value);
+    }
+    if (!m_sidreGroup->hasView("value")) {
+      m_sidreGroup->createViewScalar("value", value);
+    }
+  }
+  
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::defaultValue(double value) {
+  if (m_type != axom::sidre::DataTypeId::DOUBLE_ID) {
+    SLIC_WARNING("Field value type did not match DOUBLE");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  if (m_sidreGroup->hasView("defaultValue")) {
+    std::string msg = fmt::format("Inlet Field has already defined default value: {0}",
+                                   m_sidreGroup->getName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    if (m_docEnabled) {
+      m_sidreGroup->createViewScalar("defaultValue", value);
+    }
+    if (!m_sidreGroup->hasView("value")) {
+      m_sidreGroup->createViewScalar("value", value);
+    }
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::range(double startVal, double endVal) {
+  if (m_type != axom::sidre::DataTypeId::DOUBLE_ID) {
+    SLIC_WARNING("Field value type did not match DOUBLE");
+    setWarningFlag(m_sidreRootGroup);
+  }
+
+  if (m_sidreGroup->hasView("range") || m_sidreGroup->hasView("validValues")) {
+    std::string msg = fmt::format("Inlet Field has already defined range: {0}",
+                                   m_sidreGroup->getName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    double tuple[2];
+    tuple[0] = startVal;
+    tuple[1] = endVal;
+    auto view = m_sidreGroup->createViewAndAllocate("range", axom::sidre::DOUBLE_ID, 2);
+    view->getBuffer()->copyBytesIntoBuffer(tuple, 2*sizeof(double));
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::range(int startVal, int endVal) {
+  if (m_type != axom::sidre::DataTypeId::INT_ID) {
+    SLIC_WARNING("Field value type did not match INT");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  
+  if (m_sidreGroup->hasView("range") || m_sidreGroup->hasView("validValues")) {
+    std::string msg = fmt::format("Inlet Field has already defined range: {0}",
+                                   m_sidreGroup->getName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    int tuple[2];
+    tuple[0] = startVal;
+    tuple[1] = endVal;
+    auto view = m_sidreGroup->createViewAndAllocate("range", axom::sidre::INT_ID, 2);
+    view->getBuffer()->copyBytesIntoBuffer(tuple, 2*sizeof(int));
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::validValues(std::vector<int> set) {
+  if (m_type != axom::sidre::DataTypeId::INT_ID) {
+    SLIC_WARNING("Field value type did not match INT");
+    setWarningFlag(m_sidreRootGroup);
+  }
+
+  if (m_sidreGroup->hasView("range") || m_sidreGroup->hasView("validValues")) {
```

Comment:
This seems to prefer the first specified range (or default).
A different approach would be to overwrite an existing value, i.e. to use the last specified range.

Reply:
Note: There might be cases where one would want to allow overwriting a range.
(although, I don't have any on hand at the moment and this can always be improved in the future).

Reply:
Let's hold off on this.  
After speaking to @white238 offline, this sounds like a possible feature for the future if/when this case arises.

Reply:
#312 so we don't forget.